src/components/AccountSwitcher.tsx

```diff
@@ -0,0 +1,146 @@
+import {Str} from 'expensify-common';
+import React, {useRef, useState} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
+import useLocalize from '@hooks/useLocalize';
+import usePermissions from '@hooks/usePermissions';
+import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {connect} from '@libs/actions/Delegate';
+import {getPersonalDetailByEmail} from '@libs/PersonalDetailsUtils';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import Avatar from './Avatar';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import MenuItem from './MenuItem';
+import type {MenuItemProps} from './MenuItem';
+import Popover from './Popover';
+import {PressableWithFeedback} from './Pressable';
+import Text from './Text';
+
+function AccountSwitcher() {
+    const currentUserPersonalDetails = useCurrentUserPersonalDetails();
+    const avatarUrl = currentUserPersonalDetails?.avatar ?? '';
+    const accountID = currentUserPersonalDetails?.accountID ?? -1;
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const theme = useTheme();
+    const {canUseNewDotCopilot} = usePermissions();
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
+    const [shouldShowDelegateMenu, setShouldShowDelegateMenu] = useState(false);
+
+    const buttonRef = useRef<HTMLDivElement>(null);
+
+    const delegates = account?.delegatedAccess?.delegates ?? [];
+    const shouldShowDelegates = delegates.length > 0 && canUseNewDotCopilot;
```

Comment:
The account switcher should be looking at `delegators` not `delegates` - delegators are accounts that you can copilot into, delegates are the accounts who can copilot into you

Reply:
oops